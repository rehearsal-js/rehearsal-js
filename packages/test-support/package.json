{
  "name": "@rehearsal/test-support",
  "version": "3.0.1-beta",
  "description": "Test support utilities for rehearsal",
  "license": "BSD-2-Clause",
  "type": "module",
  "exports": {
    ".": {
      "types": "./dist/src/index.d.ts",
      "import": "./dist/src/index.js"
    },
    "./*": "./*"
  },
  "main": "dist/src/index.js",
  "types": "dist/src/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "tsc -b",
    "fixtures:clean": "rm -rf ./fixtures/ember/*/node_modules",
    "fixtures:prepare": "node ./scripts/setup-fixtures.js ./fixtures/ember/app-template ./fixtures/ember/addon-template ./fixtures/ember/app-ts-template",
    "postinstall": "pnpm fixtures:prepare",
    "lint": "npm-run-all lint:*",
    "lint:tsc-src": "tsc --noEmit",
    "lint:tsc-test": "tsc --noEmit --project test/tsconfig.json",
    "test": "vitest --run --config ./vitest.config.ts --coverage",
    "test:slow": "vitest --run --config ./vitest.config.slow.ts",
    "version": "pnpm version"
  },
  "dependencies": {
    "find-up": "^6.3.0",
    "fixturify": "^3.0.0",
    "fixturify-project": "^5.2.0",
    "rimraf": "^4.4.1",
    "scenario-tester": "^2.1.2",
    "tmp": "^0.2.1"
  },
  "devDependencies": {
    "@vitest/coverage-c8": "^0.30.1",
    "ember-cli": "^4.11.0",
    "ember-data": "^4.11.3",
    "ember-source": "^4.12.0",
    "vitest": "^0.30.0"
  },
  "peerDependencies": {
    "typescript": "^5.0"
  },
  "packageManager": "pnpm@8.2.0",
  "volta": {
    "extends": "../../package.json"
  }
}
