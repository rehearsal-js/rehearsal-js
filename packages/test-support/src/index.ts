export * from './files.js';
export * from './library.js';
export * from './project.js';
export * from './scenarios.js';
