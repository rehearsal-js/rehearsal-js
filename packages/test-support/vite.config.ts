import { defineConfig } from 'vitest/config';

//eslint-disable-next-line import/no-default-export
export default defineConfig({
  test: {
    testTimeout: 100_000,
    hookTimeout: 100_000,
    watchExclude: ['package.json', '**/fixtures/**'],
    threads: false,
  },
});
