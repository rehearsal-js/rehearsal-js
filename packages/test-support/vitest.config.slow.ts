import { defineConfig } from 'vitest/config';


// explicit includes for slow test otherwise it'll run in `test`

//eslint-disable-next-line import/no-default-export
export default defineConfig({
  test: {
    testTimeout: 500_000,
    hookTimeout: 50_000,
    watchExclude: ['package.json', '**/fixtures/**'],
    include: ['./test/scenarios-app.test.ts']
  },
});
