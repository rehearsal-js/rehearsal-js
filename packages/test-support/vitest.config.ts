import { defineConfig } from 'vitest/config';

// we dont want to run these slow tests in this config
// these will all run by default in `test:slow` instead which is run in CI
const exclude = ['./test/scenarios-app.test.ts'];

//eslint-disable-next-line import/no-default-export
export default defineConfig({
  test: {
    testTimeout: 500_000,
    hookTimeout: 50_000,
    watchExclude: ['package.json', '**/fixtures/**'],
    sequence: {
      shuffle: true,
    },
    include: ['./test/**/*.test.ts'],
    exclude,
  },
});
