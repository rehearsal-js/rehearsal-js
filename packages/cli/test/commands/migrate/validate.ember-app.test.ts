import { resolve } from 'node:path';
import { rmSync, writeFileSync } from 'node:fs';
import { createFileSync } from 'fs-extra/esm';
import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
import { createLogger, format, transports } from 'winston';
import { appScenarios, clean } from '@rehearsal/test-support';
import { Scenario, PreparedApp } from 'scenario-tester';

import { cleanOutput } from '../../test-helpers/index.js';
import { runValidate } from '../../test-helpers/valdiate-test-utils.js';

const logger = createLogger({
  transports: [new transports.Console({ format: format.cli() })],
});

describe('Task: validate', () => {
  appScenarios.forEachScenario((scenario: Scenario) => {
    describe(scenario.name, () => {
      let app: PreparedApp;
      let output = '';

      beforeEach(async () => {
        app = await scenario.prepare();
        clean(app.dir);

        vi.spyOn(console, 'info').mockImplementation((chunk) => {
          output += `${chunk}\n`;
        });
        vi.spyOn(console, 'log').mockImplementation((chunk) => {
          output += `${chunk}\n`;
        });
        vi.spyOn(console, 'error').mockImplementation((chunk) => {
          output += `${chunk}\n`;
        });

        vi.spyOn(logger, 'warn').mockImplementation((chunk) => {
          output += `${chunk}\n`;
          return logger;
        });
        vi.spyOn(logger, 'error').mockImplementation((chunk) => {
          output += `${chunk}\n`;
          return logger;
        });
      });
      afterEach(() => {
        output = '';
        vi.clearAllMocks();
      });

      test('pass with package.json', async () => {
        await runValidate(app.dir, logger);
        expect(cleanOutput(output, app.dir)).toMatchSnapshot();
      });

      test('error if no package.json', async () => {
        rmSync(resolve(app.dir, 'package.json'));
        await expect(() => runValidate(app.dir, logger)).rejects.toThrowError(
          `package.json does not exists`
        );
      });

      test('error if .gitignore has .rehearsal', async () => {
        const gitignore = `.rehearsal\nfoo\nbar`;
        const gitignorePath = resolve(app.dir, '.gitignore');
        writeFileSync(gitignorePath, gitignore, 'utf-8');

        await expect(() => runValidate(app.dir, logger)).rejects.toThrowError(
          `.rehearsal directory is ignored by .gitignore file. Please remove it from .gitignore file and try again.`
        );
      });

      test('show warning message for missing files in --regen', async () => {
        await runValidate(app.dir, logger, { regen: true });
        expect(cleanOutput(output, app.dir)).toMatchSnapshot();
      });

      test('pass with all config files in --regen', async () => {
        createFileSync(resolve(app.dir, '.eslintrc.js'));
        createFileSync(resolve(app.dir, 'tsconfig.json'));

        await runValidate(app.dir, logger, { regen: true });
        expect(cleanOutput(output, app.dir)).toMatchSnapshot();
      });
    });
  });
});
