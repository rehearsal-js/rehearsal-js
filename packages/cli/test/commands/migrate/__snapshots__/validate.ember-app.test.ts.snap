// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Task: validate > ember3-app > pass with all config files in --regen 1`] = `
"[STARTED] Validate project
[SUCCESS] Validate project"
`;

exports[`Task: validate > ember3-app > pass with package.json 1`] = `
"[STARTED] Validate project
[SUCCESS] Validate project"
`;

exports[`Task: validate > ember3-app > show warning message for missing files in --regen 1`] = `
"[STARTED] Validate project
Eslint config (.eslintrc.{js,yml,json,yaml}) does not exist. You need to run rehearsal migrate first before you can run rehearsal migrate --regen
[SUCCESS] Validate project"
`;

exports[`Task: validate > ember3-appWithInRepoAddon > pass with all config files in --regen 1`] = `
"[STARTED] Validate project
[SUCCESS] Validate project"
`;

exports[`Task: validate > ember3-appWithInRepoAddon > pass with package.json 1`] = `
"[STARTED] Validate project
[SUCCESS] Validate project"
`;

exports[`Task: validate > ember3-appWithInRepoAddon > show warning message for missing files in --regen 1`] = `
"[STARTED] Validate project
Eslint config (.eslintrc.{js,yml,json,yaml}) does not exist. You need to run rehearsal migrate first before you can run rehearsal migrate --regen
[SUCCESS] Validate project"
`;

exports[`Task: validate > ember3-appWithInRepoEngine > pass with all config files in --regen 1`] = `
"[STARTED] Validate project
[SUCCESS] Validate project"
`;

exports[`Task: validate > ember3-appWithInRepoEngine > pass with package.json 1`] = `
"[STARTED] Validate project
[SUCCESS] Validate project"
`;

exports[`Task: validate > ember3-appWithInRepoEngine > show warning message for missing files in --regen 1`] = `
"[STARTED] Validate project
Eslint config (.eslintrc.{js,yml,json,yaml}) does not exist. You need to run rehearsal migrate first before you can run rehearsal migrate --regen
[SUCCESS] Validate project"
`;
